From fad6d134f6174048353a56fedc5cbec7b8c163a5 Mon Sep 17 00:00:00 2001
From: Shawn8901 <shawn8901@googlemail.com>
Date: Wed, 30 Mar 2022 19:38:28 +0200
Subject: [PATCH] Added a cmake option control if a placeholder file should be
 witten install phase

This option enables the possibility for a package maintainer to let RTTR_GAMEDIR to a place
which is not in control of the packaging system. Thats needed when RTTR_GAMEDIR should point
to a directory which is not the package, but the packaging system prevents writes on package build time.

As the usecase is special for some packaging systems and not indended for daily use its marked as advanced option.
---
 CMakeLists.txt | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4562487ef..688d873bb 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -384,7 +384,11 @@ if(NOT WIN32)
 endif()
 
 # Placeholder for S2 installation
-install(FILES "${RTTR_S2_PLACEHOLDER_PATH}" DESTINATION "${RTTR_GAMEDIR}")
+option(RTTR_INSTALL_PLACEHOLDER "Install a placeholder file to the location the S2 game files should be copied to." ON)
+mark_as_advanced(RTTR_INSTALL_PLACEHOLDER)
+if(RTTR_INSTALL_PLACEHOLDER)
+    install(FILES "${RTTR_S2_PLACEHOLDER_PATH}" DESTINATION "${RTTR_GAMEDIR}")
+endif()
 
 ################################################################################
 # Postbuild
-- 
2.35.1
